dig/dig.go

```diff
@@ -85,12 +85,16 @@ func (g *Graph) Register(c interface{}) error {
 // Provided object must be a pointer
 // Any dependencies of the object will receive constructor calls, or be initialized (once)
 // Constructor with return value *object will be called
-//
-// TODO(glib): catch any and all panics from this method, as there is a lot of reflect going on
-func (g *Graph) Resolve(obj interface{}) error {
+func (g *Graph) Resolve(obj interface{}) (err error) {
 	g.Lock()
 	defer g.Unlock()
 
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("panic during Resolve %v", r)
```

Comment:
note that I had to change to using the named return, since that is the only way to return an error after panic recovery